### PR TITLE
Fix AzurePowerShell V5 cleanup failures overriding task result

### PR DESCRIPTION
**Pull request by manolerazvan**
### **Context**
Fixes false `Failed` results from the AzurePowerShellV5 task when its cleanup step encounters problems. The `finally` block currently calls `tl.which('pwsh', true)` which throws when neither `pwsh` nor `powershell` is on PATH, and any non-zero cleanup exit code calls `tl.setResult(Failed, 'Cleanup failed with ...')`, **overriding the actual task result**.

Real-world impact (incident 759642180):
Linked work item: [AB#2366919](https://dev.azure.com/mseng/b924d696-3eae-4116-8443-9a18392d8544/_workitems/edit/2366919)

**Precedent:** The Windows handler (`azurepowershell.ps1` line 221) has long used `Disconnect-AzureAndClearContext -ErrorAction SilentlyContinue`, treating cleanup as best-effort. This PR aligns the Node handler with that proven Windows behavior.

---

### **Task Name**
`AzurePowerShellV5` (and the auto-generated `AzurePowerShellV5_Node24` variant)

---

### **Description**
- **Cache `resolvedPwshPath`** from the main `run()` lookup so the `finally` block reuses the exact same resolution and never re-throws on its own.
- **Skip cleanup with `tl.debug(...)`** if `resolvedPwshPath` is empty — preserves the original error from `run()` (e.g., "Unable to locate executable file: 'pwsh'") as the task's final result.
- **Replace `tl.setResult(Failed, ...)` with `tl.warning(...)`** in both cleanup failure paths (non-zero exit and thrown exception). The task's actual result is never overridden by cleanup outcome.
- **Best-effort env clear**: `delete process.env.AZURESUBSCRIPTION_*` when cleanup didn't run cleanly, so subsequent steps don't pick up stale service-connection state.
- **CustomerIntelligence telemetry**: emit `##vso[telemetry.publish area=TaskHub;feature=AzurePowerShellV5_Cleanup]` with `Outcome` (`Success`/`NonZeroExit`/`Threw`/`SkippedPwshNotResolved`), `ExitCode`, truncated `ErrorMessageShort`, `AgentOS`, `AgentVersion`. Allows monitoring the new code path and observing whether cleanup is still actually failing under the hood after the warning conversion.
- Add 2 L0 mock tests covering: cleanup script exits non-zero (task still succeeds with warning) and pwsh not on PATH (original error preserved, cleanup skipped).
- Bump V5 task version to `5.276.2` and regenerate `_generated/AzurePowerShellV5{,_Node24}`.

---

### **Risk Assessment** (Low / Medium / High)
**Low.**
- Behavior matches the Windows handler that has shipped this way for years without incident.
- No new dependencies; no public input changes; no schema changes.
- Intentional behavior change: pipelines that were failing ONLY due to cleanup will now succeed with a warning. This is the desired correction.
- Possible side-effect: a downstream task with `condition: failed()` that was triggering on cleanup-only failures will no longer trigger. Highly unusual pattern.
- Customer monitoring on the substring `"Cleanup failed"` in timeline issues will no longer match — we mitigate with new CustomerIntelligence telemetry that supports both the old and new monitor query patterns.

---

### **Change Behind Feature Flag** (Yes / No)
**No.**
1. Windows handler ships the same behavior unconditionally; there's no precedent for gating it.
2. Rollback is already available via task-version pinning (`5.276.0`).

---

### **Documentation Changes Required** (Yes/No)
No.

---

### **Unit Tests Added or Updated** (Yes / No)
**Yes** — 2 new L0 mock-runner tests in `Tasks/AzurePowerShellV5/Tests/`:
- `L0Cleanup_CleanupFailsWarnsButTaskSucceeds.ts` — main script exits 0, cleanup exits 1 ⇒ task succeeds with a warning. Asserts the old `"Cleanup failed with exit code:"` Failed message is NOT emitted.
- `L0Cleanup_PwshNotFound.ts` — `tl.which('pwsh')` and `tl.which('powershell')` both empty ⇒ task fails with the original error and cleanup is skipped. Asserts the old `"Cleanup failed with error message:"` Failed message is NOT emitted.

All 11 tests in the V5 suite pass (9 pre-existing PSRunner tests + 2 new).

---

### **Additional Testing Performed**
- `node make.js build --task AzurePowerShellV5 --BypassNpmAudit` — both `AzurePowerShellV5` and `AzurePowerShellV5_Node24` build successfully.
- `npx mocha _build/Tasks/AzurePowerShellV5/Tests/L0.js --timeout 30000` — 11/11 passing.

---

### **Logging Added/Updated** (Yes/No)
**Yes** — added `tl.debug('Skipping cleanup: PowerShell executable was not resolved during main execution.')`, `tl.debug('Cleanup exit code: ...')`, and `tl.debug('Clearing service connection environment variables from agent process.')`. Cleanup failures use `tl.warning(...)`. No sensitive data is logged. Telemetry payload is bounded.

---

### **Telemetry Added/Updated** (Yes/No)
**Yes.** New CustomerIntelligence event:
- Area: `TaskHub`, Feature: `AzurePowerShellV5_Cleanup`
- Payload: `{ Outcome, ExitCode, ErrorMessageShort, AgentOS, AgentVersion, TaskVersion }`

**The existing Kusto monitor must be updated post-deploy** — the previous query matched `ErrorMessage has "Cleanup failed"` in `ClientTrace`, which this fix eliminates. Replace it so we can keep observing the actual cleanup-failure rate (now surfaced as `NonZeroExit`/`Threw`/`SkippedPwshNotResolved`).

---

### **Rollback Scenario and Process** (Yes/No)
**Yes.** Republish prior task version `5.276.0`. Operators can pin pipelines to that version while we investigate any regression. No FF or environment override needed.

---

### **Dependency Impact Assessed and Regression Tested** (Yes/No)
**Yes.** No package dependency changes. No public task-input changes. The `_generated/AzurePowerShellV5_Node24` variant is regenerated by `BuildConfigGen` from the same `task.json` and inherits the fix automatically.

---

### **Checklist**
- [x] Related issue linked ([AB#2366919](https://dev.azure.com/mseng/b924d696-3eae-4116-8443-9a18392d8544/_workitems/edit/2366919), ICM 759642180)
- [x] Task version was bumped (5.276.0 → 5.276.2 — Node24 variant becomes 5.276.3)
- [x] Verified the task behaves as expected (build green, all L0 tests passing)

### CODE REVIEW

File: Tasks/AzurePowerShellV5/task.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
   "version": {
     "Major": 5,
     "Minor": 276,
-    "Patch": 0
+    "Patch": 2
```
Comment (nemanjarogic) [inline Tasks/AzurePowerShellV5/task.json]: Any reason why we increased to `2` instead of `1`?

**Review by nemanjarogic — APPROVED** [review]

**Review by wawanawna — APPROVED** [review]

File: Tasks/AzurePowerShellV5/task.json (hunk 1 above)
Comment (manolerazvan) [reply Tasks/AzurePowerShellV5/task.json]: Because of the auto-generated _Node24 variant. The version pair must stay unique across variants, and Node24_1 is conventionally pinned to default_patch + 1.

Before this PR _generated/AzurePowerShellV5.versionmap.txt was:

 Default|5.276.0
 Node24_1|5.276.1

If I bumped Default to 5.276.1 it would collide with the existing Node24_1 build, so the next legal pair is .2 / .3:

 Default|5.276.2
 Node24_1|5.276.3

So every change to a task that has a Node24 variant consumes two patch slots — one for the legacy/Default runner, one for the Node 24 variant.

File: Tasks/AzurePowerShellV5/azurepowershell.ts [hunk 2]
```diff
@@ -180,35 +182,90 @@ async function run() {
...
+    try {
+        const payload = {
+            Outcome: outcome,
+            ExitCode: exitCode,
+            // First 200 chars of the error message only — error text is not sensitive but bound it
+            // defensively so a stack trace cannot blow up the telemetry record.
+            ErrorMessageShort: errorMessage ? errorMessage.substring(0, 200) : undefined,
+            AgentOS: process.env.AGENT_OS,
+            AgentVersion: process.env.AGENT_VERSION,
+            TaskVersion: '5'
```
Comment (ivanduplenskikh) [inline Tasks/AzurePowerShellV5/azurepowershell.ts]: Shouldn't we put here the real task version from task.json file?

Comment (manolerazvan) [reply Tasks/AzurePowerShellV5/azurepowershell.ts]: Good catch. Replaced the hardcoded '5' with getTaskVersion(), which reads Major.Minor.Patch from task.json (falls back to '5' on any read error, telemetry never throws). Now reports the real 5.276.x so Default vs Node24 builds are distinguishable. Mirrored into both _generated variants. Pushed in 8ac8312.

**Review by ivanduplenskikh — APPROVED** [review]
